src/NuGet.Core/NuGet.Configuration/Settings/Items/StoreClientCertItem.cs

```diff
@@ -0,0 +1,249 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Xml.Linq;
+
+namespace NuGet.Configuration
+{
+    /// <summary>
+    ///     A StoreClientCertItem have 4 Attributes:
+    ///     - [Required] packageSource
+    ///     - [Optional] storeLocation. StoreLocation.CurrentUser by default.
+    ///     - [Optional] storeName. StoreName.My by default.
+    ///     - [Optional] findBy. X509FindType.FindByThumbprint by default.
+    ///     - [Required] findValue
+    /// </summary>
+    public sealed class StoreClientCertItem : ClientCertItem
+    {
+        private const X509FindType DefaultFindBy = X509FindType.FindByThumbprint;
+        private const StoreLocation DefaultStoreLocation = StoreLocation.CurrentUser;
+        private const StoreName DefaultStoreName = StoreName.My;
+
+        public static string GetString(X509FindType type)
+        {
+            return type.ToString().Replace("FindBy", string.Empty);
+        }
+
+        public static string GetString(StoreName storeName)
+        {
+            return storeName.ToString();
+        }
+
+        public static string GetString(StoreLocation storeLocation)
+        {
+            return storeLocation.ToString();
+        }
+
+        public StoreClientCertItem(string packageSource,
+                             string findValue,
+                             StoreLocation? storeLocation = null,
+                             StoreName? storeName = null,
+                             X509FindType? findBy = null)
+            : base(packageSource)
+        {
+            if (!storeLocation.HasValue) storeLocation = DefaultStoreLocation;
+            if (!storeName.HasValue) storeName = DefaultStoreName;
+            if (!findBy.HasValue) findBy = DefaultFindBy;
```

Comment:
I don't like it, but the team doesn't use single line if blocks. Please add in the `{` and `}` on new lines, as you did with the null or white space check below.

Reply:
fixed